policy_parser.py

```diff
@@ -14,35 +14,140 @@ def __init__(self, msg=None, policy=None):
 
 
 class PolicyParser(object):
+    """
+    Parser for Kubernetes NetworkPolicy API objects.
+
+    They are defined by the NetworkPolicySpec in the /apis/networking API
+    group.
+    """
     def __init__(self, policy):
-        self.namespace = policy["metadata"]["namespace"]
+        """
+        Create a Parser for a Kubernetes NetworkPolicy API object.
+
+        Returns the parser.
+        Throws PolicyError if the Policy is invalid.
+        """
         self.policy = policy
+        try:
+            self.namespace = self.policy["metadata"]["namespace"]
+        except KeyError:
+            raise PolicyError(
+                "Policy object missing required fields metadata/namespace",
+                self.policy
+            )
+
+    def calculate_pod_selector(self):
+        """
+        Generate the Calico pod selector for this Policy.
+
+        Returns the pod selector in the Calico datamodel format.
+        Throws PolicyError if the Policy is invalid or unsupported.
+        """
+        _log.debug("Calculating pod selector")
+        try:
+            k8s_selector = self.policy["spec"]["podSelector"]
+        except KeyError:
+            raise PolicyError(
+                "Policy object missing required fields spec/podSelector",
+                self.policy
+            )
+
+        # PodSelectors only select pods from the Policy's namespace.
+        calico_selectors = ["%s == '%s'" %
+            (K8S_NAMESPACE_LABEL, self.namespace)]
+
+        # An empty PodSelector matches all pods in the namespace.
+        if k8s_selector:
```

Comment:
I think you can drop this `if` check - if the selector is empty, then _calculate_selectors will return something appropriate e.g `[]` (if it doesn't right now, it should!)